lib/index.js

```diff
@@ -249,16 +291,48 @@ module.exports = (secp256k1) => {
       }
     },
 
-    ecdsaSign (msg32, seckey, options = {}, output) {
+    ecdsaSign (msg32, seckey, options = {}, output, forceLegacy) {
       isUint8Array('message', msg32, 32)
       isUint8Array('private key', seckey, 32)
-      assert(toTypeString(options) === 'Object', 'Expected options to be an Object')
-      if (options.data !== undefined) isUint8Array('options.data', options.data)
-      if (options.noncefn !== undefined) assert(toTypeString(options.noncefn) === 'Function', 'Expected options.noncefn to be a Function')
+      assert(
+        toTypeString(options) === 'Object',
+        'Expected options to be an Object'
+      )
+      if (options.data !== undefined) {
+        isUint8Array('options.data', options.data)
+      }
+      if (options.noncefn !== undefined) {
+        assert(
+          toTypeString(options.noncefn) === 'Function',
+          'Expected options.noncefn to be a Function'
+        )
+      }
+
       output = getAssertedOutput(output, 64)
+      let obj = { signature: output, recid: null }
+      let res
+      if (!bitcoinerlab || !!options.noncefn || forceLegacy) {
+        res = secp256k1.ecdsaSign(
+          obj,
+          msg32,
+          seckey,
+          options.data,
+          options.noncefn
+        )
+      } else {
+        const bRes = secp256k1.ecdsaSign(msg32, seckey, options)
+        res = bRes.res
+        const that = this
+        obj = {
+          signature: bRes.signature,
+          get recid () {
+            return that.ecdsaSign(msg32, seckey, options, output, true)
```

Comment:
@mvayngrib Will this leak?

Reply:
could u elaborate?

Reply:
Do you think it will leak memory?

Reply:
No I mean since I hold on `this` do you think there might be memory leak?

Reply:
ah ic, it might. But we can avoid changing the public API and just call elliptic.ecdsaSign inside `get recid()` instead